src/oaklib/interfaces/text_annotator_interface.py

```diff
@@ -92,3 +93,16 @@ def annotate_text(
                             match_string=text[ix : ix + len(k)],
                         )
                         yield ann
+
+    def annotate_file(
```

Comment:
this is perfect

but note you don't have to repeat this in all implementations. Only override this if you don't want the default implementation

e.g an oger implementation may require a text file, so this is the right place to intercept